libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/MemoryMessageBus.scala

```diff
@@ -0,0 +1,39 @@
+package com.advancedtelematic.libats.messaging
+
+import akka.NotUsed
+import akka.actor.ActorRef
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.Source
+import com.advancedtelematic.libats.messaging.Messages.MessageLike
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class MemoryMessageBus {
+  private val _buffer = mutable.Queue.empty[Any]
+  private var _receiver = Option.empty[ActorRef]
+
+  def subscribe[T]()(implicit m: MessageLike[T]): Source[T, NotUsed] = {
+    Source.actorRef(MessageBus.DEFAULT_CLIENT_BUFFER_SIZE, OverflowStrategy.dropTail).mapMaterializedValue { ref =>
+      _receiver = Some(ref)
```

Comment:
This is actually quite shitty. It's because all this abstraction layer over source/publisher is not working well for us, we need to revisit it and implement it properly.

This actually makes the first subscriber receive all the buffer messages. If there is more than one subscriber this will actually not work properly, because yes, the ref will be changed to the last subscriber and only that subscriber will receive the messages.

The problem here is that the `mapMaterializedValue` function will only be called when something actually subscribes to that source (ie, runs the source). Which in our tests only happens after the publishing.